src/charm.py

```diff
@@ -198,6 +216,68 @@ def _get_hydra_endpoint_info(self) -> Optional[str]:
 
         return oauth2_provider_url
 
+    def _get_juju_config_identity_schema_config_(self) -> Optional[Tuple[str, Dict]]:
+        if identity_schemas := self.config.get("identity_schemas"):
+            default_schema = self.config.get("default_identity_schema")
+            if not default_schema:
+                logger.error(
+                    "`identity_schemas` configuration was set, but no `default_identity_schema` was found"
+                )
+                logger.warning("Ignoring `identity_schemas` configuration")
+                return None
+
+            try:
+                schemas = json.loads(identity_schemas)
+            except json.JSONDecodeError as e:
+                logger.error(f"identity_schemas is not a valid json: {e}")
+                logger.warning("Ignoring `identity_schemas` configuration")
+                return None
+
+            # We order the dict items to make sure the dict is ordered always has the same order
+            returned_schemas = {
+                schema_id: f"base64://{base64.b64encode(json.dumps(schema).encode()).decode()}"
+                for schema_id, schema in sorted(schemas.items())
+            }
+            return default_schema, returned_schemas
+        return None
+
+    def _get_shared_identity_schema_config(self) -> Optional[Tuple[str, Dict]]:
+        storage = self.model.storages[SHARED_IDENTITY_SCHEMA_STORAGE_NAME]
+        if not storage:
+            logger.info("No shared storage attached")
+            return
+
+        path = storage[0].location
+        if list(Path(path).glob("*.json")) != []:
+            raise NotImplementedError()
+        return None
+
+    def _get_default_identity_schema_config(self) -> Tuple[Optional[str], Optional[Dict]]:
+        logger.info("Pushing default identity schemas to container")
+        # We order the glob to make sure the dict is ordered always has the same order
```

Comment:
Again, why caring about dict ordering

Reply:
Even if it wasn't, would we care? If yes maybe we should use a different data structure